### PR TITLE
Adds simulation tests of invalidly-signed transactions.

### PR DESCRIPTION
**Pull request by joeldudley**
The test works by issuing invalid transfers alongside the valid ones during the simulation, but checking that these do not affect the simulation's stats.

### CODE REVIEW

**Review by otherview — APPROVED** [review]

**Review by tudor-malene — CHANGES_REQUESTED** [review]
Thanks Joel. This looks good but incomplete.
I don't see a check that these transactions are not included in the blockchain.

**Comment by joeldudley** [comment]
@tudor-malene We already have tests of what's included in the blockchain as part of the simulation.

If these transactions were being included (rather than rejected), it would break those various checks (e.g. balances would be wrong, transfer numbers would be wrong...).

**Comment by tudor-malene** [comment]
> @tudor-malene We already have tests of what's included in the blockchain as part of the simulation.
> 
> If these transactions were being included (rather than rejected), it would break those various checks (e.g. balances would be wrong, transfer numbers would be wrong...).

The checks are that the transactions that should be included were included. Not that there's no extra stuff.
Also, the balances are only checked holistically. That everything that was deposited minus everything that was successfully withdrawn is in the system.
There are no exact balance checks.

File: integration/simulation/transaction_manager.go [hunk 1]
```diff
@@ -137,12 +145,11 @@ func (m *TransactionManager) issueRandomTransfers() {
 			break
 		}
 		fromWallet := rndWallet(m.wallets)
-		from := fromWallet.Address
 		to := rndWallet(m.wallets).Address
-		if from == to {
-			continue
+		for fromWallet.Address == to {
+			to = rndWallet(m.wallets).Address
```
Comment (tudor-malene) [inline integration/simulation/transaction_manager.go:150]: what if it's still the same value?

Comment (tudor-malene) [inline integration/simulation/transaction_manager.go:150]: what if it's still the same value?

Comment (joeldudley) [reply integration/simulation/transaction_manager.go:150]: It'll loop until it's different.

Comment (joeldudley) [reply integration/simulation/transaction_manager.go:150]: It'll loop until it's different.

**Comment by joeldudley** [comment]
@tudor-malene It is checking, because these are now invalid withdrawals. If they were accepted with the bad sigs, it would break the balance checks in the sim (you can try modifying the code so valid sigs are produced, and you'll see it breaks the test).

File: integration/simulation/transaction_manager.go (hunk 1 above)
Comment (tudor-malene) [reply integration/simulation/transaction_manager.go:150]: ah yes, I missed that.

**Review by tudor-malene — APPROVED** [review]
LGTM.
Got it now. Since you used withdrawals, it should be included in the checks.

**Pull request merged**